### PR TITLE
Added mosaic reingestion to tgocassis2isis and errors test to tgocassisrdrgen

### PR DESCRIPTION
**Pull request by celbal**

### CODE REVIEW

File: isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp [hunk 1]
```diff
@@ -47,32 +49,29 @@ void IsisMain() {
...
-        Pvl *outputLabel = outputCube->label();
-        labelXlater.Auto(*(outputLabel));
-      } 
-      catch (IException &e) {
-        Pvl *outputLabel = outputCube->label();
-        if(outputLabel->hasGroup("Mapping")) {
-          outputLabel->deleteGroup("Mapping"); 
+      if (translateMappingLabel(xmlFileName, outputCube)) {
+        if (!translateMosaicLabel(xmlFileName, outputCube)) {
+          translateLabels(xmlFileName, outputCube, transExportFile);
```
Comment (krlberry) [inline isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp:58]: I don't remember -- did we ever address BandBin groups needing to be ingested for mosiacs, but at least when we were looking at it yesterday, only being translated in the translation files used in `translateLabels`?

Comment (krlberry) [reply isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp:58]: We could address this by, for example, adding a BandBin translation group to the Mosaic translation file.

File: isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp [hunk 2]
```diff
@@ -47,32 +49,29 @@ void IsisMain() {
     QString transRawFile = "/translations/tgoCassisInstrument.trn";
     QString transExportFile = "/translations/tgoCassisExportedInstrument.trn";
 
+
```
Comment (krlberry) [inline isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp]: Can we get rid of the extra two added lines?

File: isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp [hunk 3]
```diff
@@ -161,6 +160,108 @@ void translateCoreInfo(XmlToPvlTranslationManager labelXlater, ProcessImport &im
...
+        QString logicalIdText = logicalId.text();
+        QStringList logicalIdStringList = logicalIdText.split(":");
+        if (logicalIdStringList.contains("data_mos")) {
+          try {
+            PvlGroup &dataDir = Preference::Preferences().findGroup("DataDirectory"); 
+            QString missionDir = (QString) dataDir["Tgo"];
+            FileName mosaicTransFile(missionDir + "/translations/tgoCassisMosaic.trn");
+
+            // Get the translation manager ready for translating the mapping label
+
```
Comment (krlberry) [inline isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp]: remove line?

File: isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp (hunk 1 above)
Comment (celbal) [reply isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp:58]: No we never addressed it. I think we already have a BandBin translation file that I could use.

Comment (celbal) [reply isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp:58]: Fixed

Comment (krlberry) [reply isis/src/tgo/apps/tgocassis2isis/tgocassis2isis.cpp:58]: :+1:

**Review by krlberry — APPROVED** [review]

**Pull request merged**